docs/spark/README.md

```diff
@@ -0,0 +1,174 @@
+# Apache Spark Usage within Raster Foundry
+
+Apache Spark is a cluster computing system with a Scala API for defining execution graphs and an engine for executing them across multiple machines. GeoTrellis is a library built on top of Spark for describing geospatial transformations against raster data sets.
+
+Raster Foundry uses both to mosaic multiple large raster data sets and output the result in a way that can be consumed by a web UI.
+
+## Table of Contents
+
+* [Spark Components](#spark-components)
+  * [Master](#master)
+  * [Worker](#workers-and-executors)
+  * [Driver](#driver)
+* [Master Recovery](#master-recovery)
+* [Development Environment](#development-environment)
+  * [Building a Job JAR](#building-a-job-jar)
+  * [Local Spark Standalone Cluster](#local-spark-standalone-cluster)
+* [Safety Testing](#safety-testing)
+
+## Spark Components
+
+Often times Spark is deployed on top of a cluster resource manager, such as Apache Mesos or Apache Hadoop YARN (Yet Another Resource Negotiator). In addition to those options, Spark comes bundled with **Spark Standalone**, which is a built-in way to run Spark in a clustered environment.
```

Comment:
I don't think the usage of Spark Standalone here is a commitment to using it beyond the development environment. It was the simplest to setup locally, and provides a good base for identifying common terminology.

As far as what we use beyond the development environment, I think about it several times a week and still don't think I have a solid plan worthy of an ADR. Spark Standalone is easier to reason about, but lacks support for enabling effective multitenancy. I don't think we'll be able to get away from running a cluster manager, at least if we want to support multiple jobs with the same cluster.